analyzers/tests/SonarAnalyzer.UnitTest/PackagingTests/RuleTypeTest.cs

```diff
@@ -29,50 +29,38 @@ public class RuleTypeTest
     {
         // Rules that have been deprecated and deleted.
         // When changing this please do not forget to notify the product teams (SQ, SC, SL).
-        private readonly HashSet<string> deletedRules = new () {"S1145", "S1697", "S4142", "S2758", "S2070", "S3693", "S4432", "S2278"};
+        private static readonly HashSet<string> DeletedRules = new() { "S1145", "S1697", "S4142", "S2758", "S2070", "S3693", "S4432", "S2278" };
 
         [TestMethod]
         public void DetectRuleTypeChanges_CS() =>
-            DetectTypeChanges(csharp::SonarAnalyzer.RspecStrings.ResourceManager,
-                              CsRuleTypeMapping.RuleTypesCs,
-                              nameof(CsRuleTypeMapping.RuleTypesCs),
-                              deletedRules);
+            DetectTypeChanges(csharp::SonarAnalyzer.RspecStrings.ResourceManager, RuleTypeMappingCS.Rules, nameof(RuleTypeMappingCS));
 
         [TestMethod]
         public void DetectRuleTypeChanges_VB() =>
-            DetectTypeChanges(vbnet::SonarAnalyzer.RspecStrings.ResourceManager,
-                              VbRuleTypeMapping.RuleTypesVb,
-                              nameof(VbRuleTypeMapping.RuleTypesVb),
-                              deletedRules);
+            DetectTypeChanges(vbnet::SonarAnalyzer.RspecStrings.ResourceManager, RuleTypeMappingVB.Rules, nameof(RuleTypeMappingVB));
 
         [AssertionMethod]
-        private static void DetectTypeChanges(ResourceManager resourceManager,
-                                              IImmutableDictionary<string, string> expectedTypes,
-                                              string expectedTypesName,
-                                              ICollection<string> deletedRules)
+        private static void DetectTypeChanges(ResourceManager resourceManager, IImmutableDictionary<string, string> expectedTypes, string expectedTypesName)
         {
             var items = Enumerable.Range(1, 10000)
-                                  .Select(i => new
+                                  .Select(x => "S" + x)
+                                  .Select(x => new
                                   {
-                                      ExpectedType = expectedTypes.GetValueOrDefault(i.ToString()),
-                                      ActualType = resourceManager.GetString($"S{i}_Type"),
-                                      RuleId = i
+                                      ExpectedType = expectedTypes.GetValueOrDefault(x),
+                                      ActualType = resourceManager.GetString($"{x}_Type"),
+                                      RuleId = x
                                   })
                                   .Where(x => x.ActualType != x.ExpectedType)
                                   .ToList();
```

Comment:
The `x` changes its type in the query. Consider renaming the x to something more expressive. Also `var items` is also too generic. `RuleId` should be the first property in the anonymous type because it is the key of the list. I think this is easier to read.

```cs
var resourcesMissmatches = (from i in Enumerable.Range(1, 10000)
                            let ruleId = $"S{i}"
                            let matching = new
                            {
                                RuleId = ruleId,
                                ExpectedType = expectedTypes.GetValueOrDefault(ruleId),
                                ActualType = resourceManager.GetString($"{ruleId}_Type")
                            }
                            where matching.ActualType != matching.ExpectedType
                            select matching).ToList();
```

Reply:
The `x` is easy to follow. I've reordered the items.